src/wallet/coin_selection.rs

```diff
@@ -1339,4 +1495,140 @@ mod test {
+            let long_term_fee: i64 = LONG_TERM_FEE_RATE
+                .fee_wu(TXIN_BASE_WEIGHT + utxo.weighted_utxo.satisfaction_weight)
+                as i64;
+
+            acc + fee_rate - long_term_fee
+        });
+
+        let excess = available_value - utxos_fee - amount_needed;
+
+        // Waste without change is the excess and difference between fee and long term fee
+        let waste = Waste::calculate(&selected, None, amount_needed, fee_rate);
+
+        assert_eq!(waste.unwrap(), Waste(utxo_fee_diff + excess as i64));
+    }
+
+    #[test]
+    fn test_calculate_waste_with_change_not_set_to_none() {
+        let fee_rate = FeeRate::from_sat_per_vb(11.0);
+        let utxo_values = vec![100_000_000, 200_000_000];
+        let selected = generate_utxos_of_values(utxo_values.clone());
+        let amount_needed = 200_000_000;
+
+        // Waste without change is the excess and difference between fee and long term fee
+        let waste = Waste::calculate(&selected, Some(0), amount_needed, fee_rate);
+
+        match waste {
+            Ok(_) => assert!(false, "No change_cost means None, not Some(0)"),
+            Err(e) => match e {
+                Error::Generic(str) => {
+                    assert_eq!(str, "if there is not cost_of_change, set to None, not zero")
+                }
+                _ => assert!(
+                    false,
+                    "No error thrown. Expected not zero cost_of_change not set to None error"
+                ),
+            },
+        }
+    }
+
+    #[test]
+    fn test_calculate_waste_with_negative_timing_cost() {
+        let fee_rate = FeeRate::from_sat_per_vb(8.0);
+        let utxo_values = vec![100_000_000, 200_000_000];
+        let selected = generate_utxos_of_values(utxo_values.clone());
+        let available_value: u64 = utxo_values.iter().sum();
+        let amount_needed = 200_000_000;
+
+        let utxos: Vec<OutputGroup<'_>> = selected
+            .iter()
+            .map(|u| OutputGroup::new(u, fee_rate))
+            .collect();
+
+        let utxos_fee: u64 = utxos.clone().iter().fold(0, |acc, utxo| acc + utxo.fee);
+
+        let utxo_fee_diff: i64 = utxos.iter().fold(0, |acc, utxo| {
+            let fee_rate: i64 = utxo.fee as i64;
+            let long_term_fee: i64 = LONG_TERM_FEE_RATE
+                .fee_wu(TXIN_BASE_WEIGHT + utxo.weighted_utxo.satisfaction_weight)
+                as i64;
+
+            acc + fee_rate - long_term_fee
+        });
+
+        let excess = available_value - utxos_fee - amount_needed;
+
+        // Waste without change is the excess and difference between fee and long term fee
+        let waste = Waste::calculate(&selected, None, amount_needed, fee_rate);
+
+        assert_eq!(waste.unwrap(), Waste(utxo_fee_diff + excess as i64));
+    }
+
+    #[test]
+    fn test_calculate_waste_with_no_timing_cost_and_no_creation_cost() {
+        let fee_rate = FeeRate::from_sat_per_vb(10.0);
```

Comment:
This might be a bit more obvious:

```suggestion
        let fee_rate = FeeRate::from_sat_per_vb(LONG_TERM_FEE_RATE);
```